tests/stack/test_api.py

```diff
@@ -338,3 +338,65 @@ def test_deploy_resource(self, stack_api):
         }
         with pytest.raises(StackError):
             stack_api._deploy_resource(resource_badtype)
+
+    def test_download_workspace(self, stack_api, tmpdir):
+        """
+            stack_api._deploy_workspace should call certain workspace client functions depending
+            on object_type and error when object_type is defined incorrectly.
+        """
+        test_deploy_output = {'test': 'test'}  # default deploy_output return value
+
+        stack_api.workspace_client.client = mock.MagicMock()
+        stack_api.workspace_client.client.get_status.return_value = test_deploy_output
+        stack_api.workspace_client.export_workspace = mock.MagicMock()
+        stack_api.workspace_client.export_workspace_dir = mock.MagicMock()
+
+        test_workspace_nb_properties = TEST_WORKSPACE_NB_PROPERTIES.copy()
+        test_workspace_nb_properties.update(
+            {'source_path': os.path.join(tmpdir.strpath,
+                                         test_workspace_nb_properties['source_path'])})
+        test_workspace_dir_properties = TEST_WORKSPACE_DIR_PROPERTIES.copy()
+        test_workspace_dir_properties.update(
+            {'source_path': os.path.join(tmpdir.strpath,
+                                         test_workspace_dir_properties['source_path'])})
+
+        stack_api._download_workspace(test_workspace_dir_properties, True)
+        stack_api.workspace_client.export_workspace_dir.assert_called_once()
+        assert stack_api.workspace_client.export_workspace_dir.call_args[0][0] == \
+            test_workspace_dir_properties['path']
+        assert stack_api.workspace_client.export_workspace_dir.call_args[0][1] == \
+            test_workspace_dir_properties['source_path']
+
+        stack_api._download_workspace(test_workspace_nb_properties, True)
+        stack_api.workspace_client.export_workspace.assert_called_once()
+        assert stack_api.workspace_client.export_workspace.call_args[0][0] == \
```

Comment:
It is hard to follow `call_args[0][0]` is there a better way to do the test? (is this the standard way to to unit tests in Databricks CLI ?)

Reply:
this is how it's been done for the other unit tests. I can try finding a more understandable way to do it.